packages/graphql/src/schema/get-definition-nodes.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Debug from "debug";
+import {
+    DirectiveDefinitionNode,
+    DocumentNode,
+    EnumTypeDefinitionNode,
+    InputObjectTypeDefinitionNode,
+    InterfaceTypeDefinitionNode,
+    Kind,
+    ObjectTypeDefinitionNode,
+    ScalarTypeDefinitionNode,
+    UnionTypeDefinitionNode,
+} from "graphql";
+import { DEBUG_GENERATE } from "../constants";
+
+const debug = Debug(DEBUG_GENERATE);
+
+export type DefinitionNodes = {
+    objectTypes: ObjectTypeDefinitionNode[];
+    scalarTypes: ScalarTypeDefinitionNode[];
+    enumTypes: EnumTypeDefinitionNode[];
+    inputObjectTypes: InputObjectTypeDefinitionNode[];
+    interfaceTypes: InterfaceTypeDefinitionNode[];
+    directives: DirectiveDefinitionNode[];
+    unionTypes: UnionTypeDefinitionNode[];
+};
+
+export function getDefinitionNodes(document: DocumentNode): DefinitionNodes {
+    return document.definitions.reduce<DefinitionNodes>(
+        (definitionNodes, definition) => {
+            switch (definition.kind) {
+                case Kind.SCALAR_TYPE_DEFINITION:
+                    definitionNodes.scalarTypes.push(definition);
+                    break;
+                case Kind.OBJECT_TYPE_DEFINITION:
+                    if (!["Query", "Mutation", "Subscription"].includes(definition.name.value)) {
```

Comment:
**nitpick**: Maybe `["Query", "Mutation", "Subscription"]` could be a constant, as it is used in some other places, but mainly to give a context on why these are not pushed to objectTypes, something like:
```suggestion
                    if (!nativeTopLevelTypes.includes(definition.name.value)) {
```
-----

**suggestion(non-blocking)**: Furthermore, a small function could make this really legible (you probably can think of better naming):

```typescript
function isNativeRootType(type: DefinitionNode): boolean
```

```suggestion
                    if (!isNativeRootType(definition)) {
```

Reply:
This is now done! For now, I've left it as a util function within the file as this is the only place where it is used. We can extract it out to a utils directory if and when it gets used elsewhere.